# <p style='text-align: center;'> Data Understanding </p>

## The objectives of data understanding are :
- Understand the attributes of the data.


- Summarize the data by identifying key characteristics, such as data volume and total number of variables in the data.


- Understand the problems with the data, such as missing values, inaccuracies, and outliers.

### Here we are going to understand one dataset as "tips.csv".
- First we import the necessary Python’s libraries.


- Load the tip.csv and explore the first 5 rows of the dataset.

In [4]:
# import the required libraries.
import pandas as pd

# Load the "tips.csv" Dataset.
df = pd.read_csv("tips.csv")

# Print the first 5 records from the dataset.
df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

Explaination of each variables from the dataset :
- **total_bill :** Total bill at restaurant.
    
    
- **tip :** tip given to the waiter.
    
    
- **sex :** Gender of the person (who paid the bill).
    
    
- **smoker :** Whether the person is smoker or non-smoker.
    
    
- **day :** Which day they came to the restaurant.
    
    
- **time :** Which time they came to the restaurant.
    
    
- **size :** Total number of members came to the restaurant.

Let’s have a look at data dimensionality using shape.

In [7]:
# Print the dimensionality of the dataset.
df.shape

(244, 7)

From the output, we can see that the table contains 244 rows and 7 columns.


We can use the info() method to output some general information about the dataframe :

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


int64, float64 and object are the data types of our features. We see that 4 features are of type object, and 3 features are numeric. With this same method, we can easily see if there are any missing values. Here, there are none because each column contains 244 observations, the same number of rows we saw before with shape.

Here total_bill and tip variables are continuous data, sex, smoker, day and time variables are discrete categorical data and size variable is a discrete count data.

The describe method shows basic statistical characteristics of each numerical feature (int64 and float64 types): number of non-missing values, mean, standard deviation, median, min, max, 0.25 and 0.75 quartiles.

In [9]:
df.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

In order to see statistics on non-numerical features, one has to explicitly indicate data types of interest in the include parameter.

In [11]:
df.describe(include=["object"])

sex smoker  day    time
count    244    244  244     244
unique     2      2    4       2
top     Male     No  Sat  Dinner
freq     157    151   87     176

For categorical (type object), describe() method returns the count, unique, top (mode) and freq.

For categorical (type object) features we can use the value_counts method. Let's have a look at the distribution of sex :

In [13]:
df["sex"].value_counts()

Male      157
Female     87
Name: sex, dtype: int64

157 persons out of 244 are men and 87 are women. 


To calculate fractions, pass normalize=True to the value_counts function.

In [15]:
df["sex"].value_counts(normalize=True)

Male      0.643443
Female    0.356557
Name: sex, dtype: float64

Here 64.34% are mens and 35.66% are womens are their in this dataset.

Now use the value_counts method on smoker variable.

In [16]:
df["smoker"].value_counts()

No     151
Yes     93
Name: smoker, dtype: int64

151 persons out of 244 are non-smokers and 93 are smokers.


o calculate fractions, pass normalize=True to the value_counts function.

In [17]:
df["smoker"].value_counts(normalize=True)

No     0.618852
Yes    0.381148
Name: smoker, dtype: float64

Here 61.89% are non-smokers and 38.11% are smokers are their in this dataset.